cli/src/cmd/run.rs

```diff
@@ -89,168 +88,127 @@ impl Cmd for RunArgs {
-        let vicinity = evm_opts.vicinity()?;
-        let backend = evm_opts.backend(&vicinity)?;
-
-        // need to match on the backend type
-        let result = match backend {
-            BackendKind::Simple(ref backend) => {
-                let runner = ContractRunner::new(
-                    &evm_opts,
-                    &cfg,
-                    backend,
-                    &abi,
-                    bytecode,
-                    Some(evm_opts.sender),
-                    None,
-                    &predeploy_libraries,
-                );
-                runner.run_test(&func, needs_setup, Some(&known_contracts))?
-            }
-            BackendKind::Shared(ref backend) => {
-                let runner = ContractRunner::new(
-                    &evm_opts,
-                    &cfg,
-                    backend,
-                    &abi,
-                    bytecode,
-                    Some(evm_opts.sender),
-                    None,
-                    &predeploy_libraries,
-                );
-                runner.run_test(&func, needs_setup, Some(&known_contracts))?
-            }
+        let mut builder = ExecutorBuilder::new()
+            .with_cheatcodes(evm_opts.ffi)
+            .with_config(evm_opts.evm_env())
+            .with_spec(crate::utils::evm_spec(&config.evm_version));
+        if let Some(ref url) = self.evm_opts.fork_url {
+            let fork = Fork { url: url.clone(), pin_block: self.evm_opts.fork_block_number };
+            builder = builder.with_fork(fork);
+        }
+        if verbosity >= 3 {
+            builder = builder.with_tracing();
+        }
+        if evm_opts.debug {
+            builder = builder.with_tracing().with_debugger();
+        }
+
+        let mut result = {
+            let mut runner =
+                Runner::new(builder.build(), evm_opts.initial_balance, evm_opts.sender);
+            let (address, mut result) =
+                runner.setup(&predeploy_libraries, bytecode, needs_setup)?;
+
+            let RunResult {
+                success,
+                gas_used,
+                logs,
+                traces,
+                debug: run_debug,
+                labeled_addresses,
+                ..
+            } = runner
+                .run(address, encode_args(&IntoFunction::into(self.sig), &self.args)?.into())?;
+
+            result.success &= success;
+
+            result.gas_used = gas_used;
+            result.logs.extend(logs);
+            result.traces.extend(traces);
+            result.debug = run_debug;
+            result.labeled_addresses.extend(labeled_addresses);
+
+            result
         };
 
+        // Identify addresses in each trace
+        let local_identifier = LocalTraceIdentifier::new(&known_contracts);
+        let mut decoder = CallTraceDecoder::new_with_labels(result.labeled_addresses.clone());
+        for (_, trace) in &mut result.traces {
+            decoder.identify(trace, &local_identifier);
+        }
```

Comment:
We always identify addresses in traces for both the debugger and displaying of traces. `result.traces` is empty if tracing is turned off